nltk/parse/corenlp.py

```diff
@@ -66,7 +66,7 @@ def __init__(
         # find the most recent code and model jar
         stanford_jar = max(
             jars,
-            key=lambda model_name: re.match(self._JAR, model_name)
+            key=lambda model_name: re.search(self._JAR, model_name).group()
```

Comment:
`.search()` might return `None` it might be an issue.

(Ha, `.match()` is looking at the beginning of a string.)

Also, it looks for the 'largest' string, while the comment says about the most recent version (a number).

I think---since I strongly believe that the usage of `CoreNLPServer` should be limited to unit tests and maybe to classroom demos where the environment is controlled---the code should be as simple as possible. Probably just getting the first element from the list of `jars` is enough.

The CoreNLP java server should be started outside of a Python process.